AdyenNetworking/APIClient/APIClient.swift

```diff
@@ -32,6 +32,17 @@ public protocol APIClientProtocol: AnyObject {
     
 }
 
+/// :nodoc:
+/// Describes any async API Client.
+@available(iOS 15.0.0, *)
+public protocol AsyncAPIClientProtocol: AnyObject {
+    
+    /// :nodoc:
+    /// Performs the API request asynchronously.
+    func perform<R: Request>(_ request: R) async -> Result<R.ResponseType, Error>
```

Comment:
I think it makes much more sense if the function throws instead of returns a `Result` :-)